test/test_bplus_tree.cpp

```diff
@@ -431,4 +431,23 @@ TEST(BPlusTree_LeafCache)
     tree.destroy();
 }
 
+TEST(BPlusTree_UpgradeFromArray)
+{
+    Array arr(Allocator::get_default());
+    arr.create(Node::type_Normal, false, 0, 0);
+
+    for (int i = 0; i < 65; i++) {
+        arr.add(i);
+    }
+
+    BPlusTree<Int> tree(Allocator::get_default());
+    tree.init_from_ref(arr.get_ref());
+    tree.split_if_needed();
```

Comment:
this doesn't do a split because there are less than 1000 elements, can you add more tests like this checking various sizes: 999, 1000, 1001, 1000*1000 + 1 etc

Reply:
Added more sizes - but the test will still only be effective when node size is 4.